File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.cpp
using namespace std;

long long GCD(int a, int b) {
    if(a % b == 0)
        return b;
    
    return GCD(b, a % b);
}

long long solution(int w,int h) {
    long long answer = (long long)w * (long long)h;
    long long gcd = GCD(w, h);
    
    answer -= w + h - gcd;
    
    return answer;
}

File: 프로그래머스/2/62048. 멀쩡한 사각형/멀쩡한 사각형.js
function GCD(a, b) {
    if(a % b === 0)
        return b;

    return GCD(b, a % b);
}

function solution(w, h) {
    let answer = w * h;
    const gcd = GCD(w, h);
    
    answer -= w + h - gcd;
    
    return answer;
}